libs/dmlf/include/dmlf/distributed_learning/utilities/mnist_client_utilities.hpp

```diff
@@ -47,8 +48,13 @@ std::shared_ptr<fetch::dmlf::distributed_learning::TrainingClient<TensorType>> M
       10u, 10u, fetch::ml::details::ActivationType::SOFTMAX);
 
   // Initialise DataLoader
+  auto mnist_images = fetch::ml::utilities::read_mnist_images<TensorType>(images);
+  auto mnist_labels = fetch::ml::utilities::read_mnist_labels<TensorType>(labels);
+  mnist_labels      = fetch::ml::utilities::convert_labels_to_onehot(mnist_labels);
```

Comment:
Even better might be to modify the graph to use Ops:OneHot, it would save us some memory,

Reply:
Haha, @baykaner literally just requested the opposite change! Less memory but more processing; which do we care about more?

Reply:
two arguments for getting my way:
1. it simplifies the graph, which means mnist tutorials can be more concise (i.e. take a look at tensorflow mnist tutorial - they're not adding any onehot op to the graph)
2. processing is probably more important than memory in general anyway (and anyway why repeatedly process something that you can do only once!?)